### PR TITLE
Remove SES requirement for getting started with a new organization an…

### PR DESCRIPTION
**Pull request by codhah92**
…d user

### CODE REVIEW

File: content/container/decision-engine-service-container/controllers/auth.js [hunk 1]
```diff
@@ -390,81 +390,78 @@ function generateNewCredentials(req, res, next) {
...
-    organization: req.body.name,
-    email: req.body.username,
-    entitytype,
-    sendEmail: true,
-    ra: req.query.ra,
-  })
-    .then((result) => {
-      if (!result || !result.email) return next('Invalid credentials. Please confirm your organization and email.');
-      else return next();
+async function forgot(req, res, next) {
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/controllers/auth.js]: more semantic name, 'forgotPassword'

File: content/container/decision-engine-service-container/controllers/auth.js [hunk 2]
```diff
@@ -390,81 +390,78 @@ function generateNewCredentials(req, res, next) {
...
+        appname: periodic.settings.name,
+        hostname: periodic.settings.application.hostname || periodic.settings.name,
+        basepath: '/auth/sign-in',
+        url: periodic.settings.application.url,
+        protocol: periodic.settings.application.protocol,
+        first_name: req.controllerData.first_name,
+        organizations: req.controllerData.organizations,
+      },
+    };
+    let emailSend = await periodic.core.mailer.sendEmail(email);
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/controllers/auth.js]: const

Comment (aloonballoon) [reply content/container/decision-engine-service-container/controllers/auth.js]: ?

File: content/container/decision-engine-service-container/controllers/auth.js [hunk 3]
```diff
@@ -390,81 +390,78 @@ function generateNewCredentials(req, res, next) {
...
+      sendEmail: true,
+      ra: req.query.ra,
     })
-    .catch(err => {
-      logger.error('forgot password error', err);
-      next(err);
-    });
+    if (emailResult && emailResult.email && emailResult.email.aws_ses_config_error) return next('Invalid AWS SES email configuration. You need to add a valid AWS SES accessKeyId and secret to access this functionality.')
+    if (!emailResult || !emailResult.email) return next('Invalid credentials. Please confirm your organization and email.');
+    else return next();
```
Comment (cdkirkwood) [inline content/container/decision-engine-service-container/controllers/auth.js]: unnecessary else statement

File: content/container/decision-engine-service-container/controllers/user.js [hunk 4]
```diff
@@ -390,19 +387,22 @@ function getUser(req, res, next) {
   }
 }
 
-function resendInvitiationEmail(req, res, next) {
+async function resendInvitiationEmail(req, res, next) {
   req.body.username = req.controllerData.user.email;
-  return utilControllers.auth.emailUser({ user: Object.assign({}, req.controllerData.user, { company: req.controllerData.user.association.organization.name, }), ra: {}, basepath: '/auth/accept-invite', subject: 'Invitation To Join', emailtemplatefilepath: passportSettings.emails.new_user, })
-    .then(() => {
+  try {
+    const emailResult = await utilControllers.auth.emailUser({ user: Object.assign({}, req.controllerData.user, { company: req.controllerData.user.association.organization.name, }), ra: {}, basepath: '/auth/accept-invite', subject: 'Invitation To Join', emailtemplatefilepath: passportSettings.emails.new_user, })
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/controllers/user.js]: can we expand this out a bit, write it in the waterfall format, listing each key and value on a new line

File: content/container/decision-engine-service-container/controllers/user.js [hunk 5]
```diff
@@ -547,10 +547,13 @@ function changeEmail(req, res, next) {
     });
     let millisecondsPerDay = 86400000;
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/controllers/user.js]: const

File: content/container/decision-engine-service-container/controllers/user.js [hunk 6]
```diff
@@ -547,10 +547,13 @@ function changeEmail(req, res, next) {
...
     if (user.extensionattributes && user.extensionattributes.passport && user.extensionattributes.passport.user_activation_token && user.extensionattributes.passport.user_activation_token_link && user.extensionattributes.passport.reset_activation_expires_millis && (user.extensionattributes.passport.reset_activation_expires_millis > (new Date().getTime() - millisecondsPerDay))) {
-      return utilControllers.auth.emailUser({ user: user, basepath: '/auth/complete_registration', subject: 'Verify Your Email Address', emailtemplatefilepath: passportSettings.emails.welcome, })
-        .then(() => {
+      const emailResult = await utilControllers.auth.emailUser({ user: user, basepath: '/auth/complete_registration', subject: 'Verify Your Email Address', emailtemplatefilepath: passportSettings.emails.welcome, })
+      try {
+        if (emailResult && emailResult.aws_ses_config_error) {
+          res.status(500).send({ message: 'Invalid AWS SES email configuration. You need to add a valid AWS SES accessKeyId and secret to access this functionality.' });
+        } else {
           let redisClient = periodic.app.locals.redisClient;
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/controllers/user.js]: const

File: content/container/decision-engine-service-container/controllers/user.js [hunk 7]
```diff
@@ -533,7 +533,7 @@ function formatUser(req, res, next) {
  * @param {Object} res Express response object.
  * @param {function} next express next function
  */
-function changeEmail(req, res, next) {
+async function changeEmail(req, res, next) {
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/controllers/user.js]: refactor function

File: content/container/decision-engine-service-container/controllers/user.js [hunk 8]
```diff
@@ -569,19 +571,20 @@ function changeEmail(req, res, next) {
...
-          return utilControllers.auth.emailUser({ user: validatedUser, basepath: '/auth/complete_registration', subject: 'Verify Your Email Address', emailtemplatefilepath: passportSettings.emails.welcome, });
-        })
-        .then(() => {
+      const validatedUser = await passportUtilities.token.generateUserActivationData({ user: req.controllerData.user, })
+      const emailResult = await utilControllers.auth.emailUser({ user: validatedUser, basepath: '/auth/complete_registration', subject: 'Verify Your Email Address', emailtemplatefilepath: passportSettings.emails.welcome, })
+      try {
+        if (emailResult && emailResult.aws_ses_config_error) {
+          res.status(500).send({ message: 'Invalid AWS SES email configuration. You need to add a valid AWS SES accessKeyId and secret to access this functionality.' });
+        } else {
           let redisClient = periodic.app.locals.redisClient;
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/controllers/user.js]: ah hem

File: content/container/decision-engine-service-container/utilities/controllers/auth.js [hunk 9]
```diff
@@ -342,48 +351,34 @@ function invalidateToken(options) {
...
+  try {
+    const User = periodic.datas.get('standard_user');
+    let emailResult = {};
+    let updatedUserAccount = {};
+    const updatedUser = await invalidateToken({ user });
+    updatedUser.password = req.body.password;
+    updatedUser[ passportSettings.registration.matched_password_field ] = req.body[ passportSettings.registration.matched_password_field ];
+    updatedUserAccount = updatedUser;
+    const validatedUser = await passportUtilities.account.validate({ user: updatedUser, });
+    updatedUserAccount = validatedUser;
```
Comment (cdkirkwood) [inline content/container/decision-engine-service-container/utilities/controllers/auth.js]: what is this mess?

File: content/container/decision-engine-service-container/utilities/controllers/auth.js [hunk 10]
```diff
@@ -342,48 +351,34 @@ function invalidateToken(options) {
...
-            email: emailStatus,
-            user: updatedUserAccount,
-          });
-        })
-        .catch(reject);
-    } catch (e) {
-      reject(e);
+  try {
+    const User = periodic.datas.get('standard_user');
+    let emailResult = {};
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/utilities/controllers/auth.js]: const

Comment (aloonballoon) [reply content/container/decision-engine-service-container/utilities/controllers/auth.js]: const

File: content/container/decision-engine-service-container/utilities/controllers/auth.js [hunk 11]
```diff
@@ -342,48 +351,34 @@ function invalidateToken(options) {
...
+    let updatedUserAccount = {};
```
Comment (aloonballoon) [inline content/container/decision-engine-service-container/utilities/controllers/auth.js]: const

File: content/container/decision-engine-service-container/controllers/user.js (hunk 8 above)
Comment (aloonballoon) [reply content/container/decision-engine-service-container/controllers/user.js]: const

**Pull request merged**